### PR TITLE
Add bindings for TimePickerAndroid

### PR DESCRIPTION
**Pull request by rodan-lewarx**
Create bindings for TimePickerAndroid based on the bindings for DatePickerAndroid. I used the same conventions used there, but bs.deriving.abstract should probably be used in favor of deprecated . objects. 

Also, I have hours and minutes typed as ints, should I type them as variants? Hour is in fact 0-23 and minute is 0-59.

### CODE REVIEW

File: src/timePickerAndroid.re [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+type responseJs = {
+  .
+  "action": string,
+  "hour": int,
+  "minute": int,
+};
+
+let action = resp =>
+  if (resp##action == timeSet) {
```
Comment (czystyl) [inline src/timePickerAndroid.re]: What do you thinking about to use pattermatching here? Like this:
```
switch (resp##action){
| timeSet => Set({hour: resp##hour, minute: resp##minute})
| dismissed => Dismissed
| _ => failwith()
}
```

File: src/timePickerAndroid.re [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+type mode =
+  | Clock
+  | Spinner
+  | Default;
+
+type action =
+  | Dismissed
+  | Set(response);
+
+type optsJs = {
```
Comment (czystyl) [inline src/timePickerAndroid.re]: We would like to use `@bs.deriving` where you need to transform to JS object.

File: src/timePickerAndroid.re [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  .
+  "hour": Js.Nullable.t(int),
+  "minute": Js.Nullable.t(int),
+  "is24Hour": Js.Nullable.t(bool),
+  "mode": string,
+};
+
+type responseJs = {
```
Comment (czystyl) [inline src/timePickerAndroid.re]: The same here, try to use `@bs.deriving`.

File: src/timePickerAndroid.re (hunk 2 above)
Comment (rodan-lewarx) [reply src/timePickerAndroid.re]: Updated the PR to deriving abstract and deriving jsConverter

File: src/timePickerAndroid.re (hunk 1 above)
Comment (rodan-lewarx) [reply src/timePickerAndroid.re]: This didn't work for me :-(. It seems you can compare to external or let variables in switch like this.

**Comment by rodan-lewarx** [comment]
@czystyl I have updated the PR using bs.deriving abstract and bs.deriving jsConverter. I have also handled the range checks for minutes and hours.

If this is ok I can update the datePickerAndroid in a similar way.

**Comment by czystyl** [comment]
@rodan-lewarx Thanks for the update! I'll look into that as soon as possible 👍

**Pull request merged**